test/filter.test.ts

```diff
@@ -24,6 +26,9 @@ describe("FilterCheck", () => {
 		expect(() => FilterCheck(Never)(null)).toThrowError(ValidationError);
 		expect(FilterCheck(Literal("a"))("a")).toEqual("a");
 		expect(FilterCheck(Boolean)(true)).toEqual(true);
+		expect(
+			FilterCheck(BigIntT)(BigInt("9007199254740993"))
+		).toMatchInlineSnapshot(`9007199254740993n`);
```

Comment:
this is a bit awkward because we can't use BigInt literals without targeting ES2020